<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Coexpression for Transcription Factors using Regulatory Impact Factors 
  and Partial Correlation and Information Theory analysis • CeTF</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Coexpression for Transcription Factors using Regulatory Impact Factors 
  and Partial Correlation and Information Theory analysis">
<meta property="og:description" content="This package provides the necessary functions for performing the 
  Partial Correlation coefficient with Information Theory (PCIT) (Reverter and 
  Chan 2008) and Regulatory Impact Factors (RIF) (Reverter et al. 2010) 
  algorithm. The PCIT algorithm identifies meaningful correlations to define 
  edges in a weighted network and can be applied to any correlation-based 
  network including but not limited to gene co-expression networks, while the 
  RIF algorithm identify critical Transcription Factors (TF) from gene 
  expression data. These two algorithms when combined provide a very relevant 
  layer of information for gene expression studies (Microarray, RNA-seq and 
  single-cell RNA-seq data).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CeTF</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/CeTF.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="cetf-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#cetf-" class="anchor"></a>ceTF <img src="articles/logo.png" align="right" width="300">
</h1></div>
<p>An implementation of PCIT and RIF analysis in R</p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>This package provides the necessary instructions for performing the Partial Correlation coefficient with Information Theory (PCIT) from <a href="https://doi.org/10.1093/bioinformatics/btn482">Reverter and Chan 2008</a> and Regulatory Impact Factors (RIF) from <a href="https://doi.org/10.1093/bioinformatics/btq051">Reverter et al. 2010</a> algorithm. The PCIT algorithm identifies meaningful correlations to define edges in a weighted network. The algorithm can be applied to any correlation-based network including but not limited to gene co-expression networks. While the RIF algorithm identify critical transcript factors (TF) from gene expression data. These two algorithms when combined provide a very relevant layer of information for gene expression studies (Microarray, RNA-seq and single-cell RNA-seq data).</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To properly run <b>ceTF</b> package is necessary to install some dependencies. For Linux users is necessary to install the following dependencies:</p>
<ul>
<li>libcurl4-openssl-dev</li>
<li>libxml2-dev</li>
<li>libssl-dev</li>
<li>gfortran</li>
<li>build-essential</li>
<li>libz-dev</li>
<li>zlib1g-dev</li>
</ul>
<p>To install R packages dependencies, run:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#CRAN dependencies</span>
<span class="va">packagesCRAN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, <span class="st">'crayon'</span>, <span class="st">'dplyr'</span>, <span class="st">'geomnet'</span>, <span class="st">'GGally'</span>, <span class="st">'ggplot2'</span>, <span class="st">'ggpubr'</span>, <span class="st">'ggrepel'</span>, <span class="st">'kableExtra'</span>, <span class="st">'knitr'</span>, <span class="st">'network'</span>, <span class="st">'pbapply'</span>, <span class="st">'reshape2'</span>, <span class="st">'rmarkdown'</span>, <span class="st">'scales'</span>, <span class="st">'testthat'</span>, <span class="st">'tidyr'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="va">packagesCRAN</span><span class="op">[</span><span class="op">!</span><span class="va">packagesCRAN</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>

<span class="co">#Bioconductor dependencies</span>
<span class="va">packagesBioc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'airway'</span>, <span class="st">'clusterProfiler'</span>, <span class="st">'DESeq2'</span>, <span class="st">'org.Hs.eg.db'</span>, <span class="st">'SummarizedExperiment'</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="va">packagesBioc</span><span class="op">[</span><span class="op">!</span><span class="va">packagesBioc</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p>Finally, to install ceTF package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"cbiagii/ceTF"</span><span class="op">)</span></code></pre></div>
</div>
<div id="docker" class="section level2">
<h2 class="hasAnchor">
<a href="#docker" class="anchor"></a>Docker</h2>
<p>To install docker follow the instructions in the links below depending on your operating system</p>
<ul>
<li>CentOS: <a href="https://docs.docker.com/install/linux/docker-ce/centos/" class="uri">https://docs.docker.com/install/linux/docker-ce/centos/</a>
</li>
<li>Debian: <a href="https://docs.docker.com/install/linux/docker-ce/debian/" class="uri">https://docs.docker.com/install/linux/docker-ce/debian/</a>
</li>
<li>Fedora: <a href="https://docs.docker.com/install/linux/docker-ce/fedora/" class="uri">https://docs.docker.com/install/linux/docker-ce/fedora/</a>
</li>
<li>Ubuntu: <a href="https://docs.docker.com/install/linux/docker-ce/ubuntu/" class="uri">https://docs.docker.com/install/linux/docker-ce/ubuntu/</a>
</li>
<li>MacOS: <a href="https://docs.docker.com/docker-for-mac/install/" class="uri">https://docs.docker.com/docker-for-mac/install/</a>
</li>
<li>Windows: <a href="https://docs.docker.com/docker-for-windows/install/" class="uri">https://docs.docker.com/docker-for-windows/install/</a>
</li>
</ul>
<div id="docker-pull" class="section level4">
<h4 class="hasAnchor">
<a href="#docker-pull" class="anchor"></a>Docker pull</h4>
<p>Once docker is installed, the next step is pull the <strong>ceTF</strong> image from dockerhub using the following command:</p>
<pre class="docker"><code>docker pull biagii/ceTF:0.99.0</code></pre>
</div>
<div id="running-image" class="section level4">
<h4 class="hasAnchor">
<a href="#running-image" class="anchor"></a>Running image</h4>
<p>There are several differents parameters to run the downloaded image. The most commom way is executing the following command:</p>
<pre class="docker"><code>docker run --rm -d -p PORT:PORT -e PASSWORD=password --name [ANY_NAME] -v /server/path/:/docker/path/ -e USERID=$UID biagii/ceTF:0.99.0</code></pre>
<p>Once the docker image is running, the Rstudio interface with all the necessary dependencies will be made available, and of course the ceTF package, installed ready for use.</p>
</div>
</div>
<div id="help" class="section level2">
<h2 class="hasAnchor">
<a href="#help" class="anchor"></a>Help</h2>
<p>
Any questions contact the developer by email: <a href="#">biagi@usp.br</a>
</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/CeTF">https://​www.bioconductor.org/​packages/​CeTF</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing CeTF</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Carlos Alberto Oliveira de Biagi Junior <br><small class="roles"> Author, maintainer </small>  </li>
<li>Ricardo Perecin 
  Nociti <br><small class="roles"> Author </small>  </li>
<li>Breno Osvaldo Funicheli <br><small class="roles"> Author </small>  </li>
<li>Wilson Araújo da Silva Junior <br><small class="roles"> Author, thesis advisor </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Carlos Alberto Oliveira de Biagi Junior, Ricardo Perecin 
  Nociti, Breno Osvaldo Funicheli, Wilson Araújo da Silva Junior.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
