<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing Regulatory Impact Factors and Partial Correlation and Information Theory • CeTF</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing Regulatory Impact Factors and Partial Correlation and Information Theory">
<meta property="og:description" content="CeTF">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CeTF</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CeTF.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CeTF_files/header-attrs-2.9/header-attrs.js"></script><script src="CeTF_files/kePrint-0.0.1/kePrint.js"></script><link href="CeTF_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyzing Regulatory Impact Factors and Partial Correlation and Information Theory</h1>
                        <h4 class="author">Carlos Alberto Oliveira de Biagi Junior, Ricardo Perecin Nociti, Breno Osvaldo Funicheli, João Paulo Bianchi Ximenez, Patrícia de Cássia Ruy, Marcelo Gomes de Paula, Rafael dos Santos Bezerra and Wilson Araújo da Silva Junior</h4>
            
            <h4 class="date">08/10/2021</h4>
      
      
      <div class="hidden name"><code>CeTF.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette provides the necessary instructions for performing the Partial Correlation coefficient with Information Theory (PCIT) <span class="citation">(Reverter and Chan 2008)</span> and Regulatory Impact Factors (RIF) <span class="citation">(Reverter et al. 2010)</span> algorithm.</p>
<p>The PCIT algorithm identifies meaningful correlations to define edges in a weighted network and can be applied to any correlation-based network including but not limited to gene co-expression networks, while the RIF algorithm identify critical transcript factors (TF) from gene expression data.</p>
<p>These two algorithms when combined provide a very relevant layer of information for gene expression studies (Microarray, RNA-seq and single-cell RNA-seq data).</p>
<div id="regulatory-information-factors-rif" class="section level2">
<h2 class="hasAnchor">
<a href="#regulatory-information-factors-rif" class="anchor"></a>Regulatory Information Factors (RIF)</h2>
<p>A gene expression data from microarray, RNA-seq or single-cell RNA-seq spanning two biological conditions of interest (e.g. normal/tumor, healthy/disease, malignant/nonmalignant) is subjected to standard normalization techniques and significance analysis to identify the target genes whose expression is differentially expressed (DE) between the two conditions. Then, the regulators (e.g. Transcript Factors genes) are identified in the data. The TF genes can be obtained from the literature <span class="citation">(Wang and Nishida 2015)</span><span class="citation">(Vaquerizas et al. 2009)</span>. Next, the co-expression correlation between each TF and the DE genes is computed for each of the two conditions. This allows for the computation of the differential wiring (DW) from the difference in co-expression correlation existing between a TF and a DE genes in the two conditions. As a result, RIF analysis assigns an extreme score to those TF that are consistently most differentially co-expressed with the highly abundant and highly DE genes (case of RIF1 score), and to those TF with the most altered ability to act as predictors of the abundance of DE genes (case of RIF2 score). A given TF may not show a change in expression profile between the two conditions to score highly by RIF as long as it shows a big change in co-expression with the DE genes. To this particular, the profile of the TF gene (triangle, solid line) is identical in both conditions (slightly downwards). Instead, the DE gene (circle, dashed line) is clearly over-expressed in condition B. Importantly, the expression of the TF and the DE gene shows a strong positive correlation in condition A, and a strong negative correlation in condition B.</p>
<center>
<img src="fig1.jpg" title="fig:" style="width:50.0%" alt="A schematic diagram of the RIF analysis. (A) Gene expression data is normalized and statistically assessed to identify differentially expressed (DE) genes and differentially PIF genes (represented by circles) which together are deemed as the Target genes; Simultaneously, (B) transcription factors (TF, represented by triangles) included in the microarray are collected and (C) their co-expression correlation with the target genes computed for each of the two conditions of interest; Finally, (D) the way in which TF and target genes are differentially co-expressed between the two conditions is used to compute the relevance of each TF according to RIF1 and RIF2 (Reverter et al. 2010).">
</center>
</div>
<div id="partial-correlation-with-information-theory-pcit" class="section level2">
<h2 class="hasAnchor">
<a href="#partial-correlation-with-information-theory-pcit" class="anchor"></a>Partial Correlation with Information Theory (PCIT)</h2>
<p>The proposed PCIT algorithm contains two distinct steps as follows:</p>
<div id="step-1---partial-correlations" class="section level4">
<h4 class="hasAnchor">
<a href="#step-1---partial-correlations" class="anchor"></a><strong>Step 1 - Partial correlations</strong>
</h4>
<p>For every trio of genes in x, y and z, the three first-order partial correlation coefficients are computed by:</p>
<p><span class="math display">\[r_{xy.z} = \frac{r_{xy} - r_{xz} r_{yz}}{\sqrt{(1-r^{2}_{xz})(1-r^{2}_{yz})}}\]</span>, and similarly for <span class="math inline">\(r_{xz.y}\)</span> and <span class="math inline">\(r_{yz.x}\)</span>.</p>
<p>The partial correlation coefficient between <em>x</em> and <em>y</em> given <em>z</em> (here denoted by <span class="math inline">\(r_{xy.z}\)</span>) indicates the strength of the linear relationship between <em>x</em> and <em>y</em> that is independent of (uncorrelated with) <em>z</em>. Calculating the ordinary (or unconditional or zero-order) correlation coefficient and comparing it with the partial correlation, we might see that the association between the two variables has been sharply reduced after eliminating the effect of the third variable.</p>
</div>
<div id="step-2---information-theory" class="section level4">
<h4 class="hasAnchor">
<a href="#step-2---information-theory" class="anchor"></a><strong>Step 2 - Information theory</strong>
</h4>
<p>We invoke the Data Processing Inequality (DPI) theorem of Information Theory which states that ‘no clever manipulation of the data can improve the inference that can be made from the data’ <span class="citation">(Cover and Thomas 2012)</span>. For every trio of genes, and in order to obtain the tolerance level (<span class="math inline">\(\varepsilon\)</span>) to be used as the local threshold for capturing significant associations, the average ratio of partial to direct correlation is computed as follows:</p>
<p><span class="math display">\[\varepsilon = (\frac{r_{xy.z}}{r_{xy}} + \frac{r_{xz.y}}{r_{xz}} + \frac{r_{yz.x}}{r_{yz}})\]</span> In the context of our network reconstruction, a connection between genes <em>x</em> and <em>y</em> is discarded if:</p>
<p><span class="math display">\[|r_{xy}| \le |\varepsilon r_{xz}| and |r_{xy}| \le |\varepsilon r_{yz}|\]</span> Otherwise, the association is defined as significant, and a connection between the pair of genes is established in the reconstruction of the GCN. To ascertain the significance of the association between genes <em>x</em> and <em>y</em>, the above mentioned Steps 1 and 2 are repeated for each of the remaining <em>n−2</em> genes (denoted here by <em>z</em>).</p>
</div>
</div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install, just type:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"cbiagii/CeTF"</span><span class="op">)</span></code></pre></div>
<p>for Linux users is necessary to install <strong>libcurl4-openssl-dev</strong>, <strong>libxml2-dev</strong> and <strong>libssl-dev</strong> dependencies.</p>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<p>There are many ways to perform the analysis. The following sections will be splited by steps, and finishing with the complete analysis with visualization. We will use the <strong>airway</strong> <span class="citation">(Himes et al. 2014)</span> dataset in the following sections. This dataset provides a RNA-seq count data from four human ASM cell lines that were treated with dexamenthasone - a potent synthetic glucocorticoid. Briefly, this dataset has 4 samples untreated and other 4 samples with the treatment.</p>
<div id="pcit" class="section level2">
<h2 class="hasAnchor">
<a href="#pcit" class="anchor"></a>PCIT</h2>
<p>The first option is to perform the PCIT analysis. The output will be a list with 3 elements. The first one contains a dataframe with the pairwise correlation between genes (corr1) and the significant pairwise correlation (corr2 <span class="math inline">\(\neq\)</span> 0). The second element of the list stores the adjacency matrix with all correlation. And the last element contains the adjacency matrix with only the significant values:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">CeTF</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>

<span class="co"># Loading airway data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span>

<span class="co"># Creating a variable with annotation data</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="va">anno</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>cond <span class="op">=</span> <span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, 
                   row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Creating a variable with count data</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>

<span class="co"># Sorting count data samples by conditions (untrt and trt)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"_untrt"</span>, <span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"_trt"</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Differential Expression analysis to use only informative genes</span>
<span class="va">DEGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/expDiff.html">expDiff</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">counts</span>,
                   anno <span class="op">=</span> <span class="va">anno</span>,
                   conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'untrt'</span>, <span class="st">'trt'</span><span class="op">)</span>,
                   lfc <span class="op">=</span> <span class="fl">4.5</span>,
                   padj <span class="op">=</span> <span class="fl">0.05</span>, 
                   diffMethod <span class="op">=</span> <span class="st">"Reverter"</span><span class="op">)</span>

<span class="co"># Selecting only DE genes from counts data</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">DEGenes</span><span class="op">$</span><span class="va">DE_unique</span><span class="op">)</span>, <span class="op">]</span>

<span class="co"># Converting count data to TPM</span>
<span class="va">tpm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">counts</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
            <span class="op">(</span><span class="fl">1e+06</span> <span class="op">*</span> <span class="va">x</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
        <span class="op">}</span><span class="op">)</span>

<span class="co"># Count normalization</span>
<span class="va">PCIT_input</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normExp.html">normExp</a></span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span>

<span class="co"># PCIT input for untrt</span>
<span class="va">PCIT_input_untrt</span> <span class="op">&lt;-</span> <span class="va">PCIT_input</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"_untrt"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">PCIT_input</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="co"># PCIT input for trt</span>
<span class="va">PCIT_input_trt</span> <span class="op">&lt;-</span> <span class="va">PCIT_input</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"_trt"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">PCIT_input</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="co"># Performing PCIT analysis for untrt condition</span>
<span class="va">PCIT_out_untrt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PCIT.html">PCIT</a></span><span class="op">(</span><span class="va">PCIT_input_untrt</span>, tolType <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span>

<span class="co"># Performing PCIT analysis for trt condition</span>
<span class="va">PCIT_out_trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PCIT.html">PCIT</a></span><span class="op">(</span><span class="va">PCIT_input_trt</span>, tolType <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span>

<span class="co"># Printing first 10 rows for untrt condition</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">PCIT_out_untrt</span><span class="op">$</span><span class="va">tab</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"striped"</span>, full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<thead><tr>
<th style="text-align:left;">
gene1
</th>
<th style="text-align:left;">
gene2
</th>
<th style="text-align:right;">
corr1
</th>
<th style="text-align:right;">
corr2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000026025
</td>
<td style="text-align:right;">
0.01160
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000035403
</td>
<td style="text-align:right;">
0.61286
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000049323
</td>
<td style="text-align:right;">
-0.71685
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000067225
</td>
<td style="text-align:right;">
-0.99935
</td>
<td style="text-align:right;">
-0.99935
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000071127
</td>
<td style="text-align:right;">
-0.56817
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000071242
</td>
<td style="text-align:right;">
-0.89063
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000075624
</td>
<td style="text-align:right;">
0.09820
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000077942
</td>
<td style="text-align:right;">
-0.80938
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000080824
</td>
<td style="text-align:right;">
-0.72639
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000087086
</td>
<td style="text-align:right;">
-0.91357
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Printing first 10 rows for trt condition</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">PCIT_out_trt</span><span class="op">$</span><span class="va">tab</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"striped"</span>, full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<thead><tr>
<th style="text-align:left;">
gene1
</th>
<th style="text-align:left;">
gene2
</th>
<th style="text-align:right;">
corr1
</th>
<th style="text-align:right;">
corr2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000026025
</td>
<td style="text-align:right;">
0.63800
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000035403
</td>
<td style="text-align:right;">
0.96857
</td>
<td style="text-align:right;">
0.96857
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000049323
</td>
<td style="text-align:right;">
0.14794
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000067225
</td>
<td style="text-align:right;">
-0.99687
</td>
<td style="text-align:right;">
-0.99687
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000071127
</td>
<td style="text-align:right;">
-0.92734
</td>
<td style="text-align:right;">
-0.92734
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000071242
</td>
<td style="text-align:right;">
-0.75914
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000075624
</td>
<td style="text-align:right;">
-0.70489
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000077942
</td>
<td style="text-align:right;">
-0.29712
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000080824
</td>
<td style="text-align:right;">
-0.93148
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000011465
</td>
<td style="text-align:left;">
ENSG00000087086
</td>
<td style="text-align:right;">
-0.94620
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Adjacency matrix: PCIT_out_untrt$adj_raw or PCIT_out_trt$adj_raw</span>
<span class="co"># Adjacency matrix only with the significant values: PCIT_out_untrt$adj_sig or PCIT_out_trt$adj_sig</span></code></pre></div>
</div>
<div id="histogram-of-connectivity-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#histogram-of-connectivity-distribution" class="anchor"></a>Histogram of connectivity distribution</h2>
<p>After performing the PCIT analysis, it is possible to verify the histogram distribution of the clustering coefficient of the adjacency matrix with the significant values:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example for trt condition</span>
<span class="fu"><a href="../reference/histPlot.html">histPlot</a></span><span class="op">(</span><span class="va">PCIT_out_trt</span><span class="op">$</span><span class="va">adj_sig</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/hist-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="density-plot-of-raw-correlation-and-significant-pcit" class="section level2">
<h2 class="hasAnchor">
<a href="#density-plot-of-raw-correlation-and-significant-pcit" class="anchor"></a>Density Plot of raw correlation and significant PCIT</h2>
<p>It’s possible to generate the density plot with the significance values of correlation. We’ll use the raw adjacency matrix and the adjacency matrix with significant values. It is necessary to define a cutoff of the correlation module (values between -1 and 1) that will be considered as significant:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example for trt condition</span>
<span class="fu"><a href="../reference/densityPlot.html">densityPlot</a></span><span class="op">(</span>mat1 <span class="op">=</span> <span class="va">PCIT_out_trt</span><span class="op">$</span><span class="va">adj_raw</span>, 
           mat2 <span class="op">=</span> <span class="va">PCIT_out_trt</span><span class="op">$</span><span class="va">adj_sig</span>, 
           threshold <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/densitySig-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="rif" class="section level2">
<h2 class="hasAnchor">
<a href="#rif" class="anchor"></a>RIF</h2>
<p>To perform the RIF analysis we will need the count data, an annotation table and a list with the Transcript Factors of specific organism (<em>Homo sapiens</em> in this case) and follow the following steps in order to get the output (dataframe with the average expression, RIF1 and RIF2 metrics for each TF):</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">CeTF</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>

<span class="co"># Loading airway data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span>

<span class="co"># Creating a variable with annotation data</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="va">anno</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>cond <span class="op">=</span> <span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, 
                   row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Creating a variable with count data</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>

<span class="co"># Sorting count data samples by conditions (untrt and trt)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"_untrt"</span>, <span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"_trt"</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Differential Expression analysis to use only informative genes</span>
<span class="va">DEGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/expDiff.html">expDiff</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">counts</span>,
                   anno <span class="op">=</span> <span class="va">anno</span>,
                   conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'untrt'</span>, <span class="st">'trt'</span><span class="op">)</span>,
                   lfc <span class="op">=</span> <span class="fl">2</span>,
                   padj <span class="op">=</span> <span class="fl">0.05</span>, 
                   diffMethod <span class="op">=</span> <span class="st">"Reverter"</span><span class="op">)</span>

<span class="co"># Selecting only DE genes from counts data</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">DEGenes</span><span class="op">$</span><span class="va">DE_unique</span><span class="op">)</span>, <span class="op">]</span>

<span class="co"># Converting count data to TPM</span>
<span class="va">tpm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">counts</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
            <span class="op">(</span><span class="fl">1e+06</span> <span class="op">*</span> <span class="va">x</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
        <span class="op">}</span><span class="op">)</span>

<span class="co"># Count normalization</span>
<span class="va">Clean_Dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normExp.html">normExp</a></span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span>

<span class="co"># Loading the Transcript Factors (TFs) character</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"TFs"</span><span class="op">)</span>

<span class="co"># Verifying which TFs are in the subsetted normalized data</span>
<span class="va">TFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Clean_Dat</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Clean_Dat</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">TFs</span><span class="op">]</span>

<span class="co"># Selecting the Target genes</span>
<span class="va">Target</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">Clean_Dat</span><span class="op">)</span>, <span class="va">TFs</span><span class="op">)</span>

<span class="co"># Ordering rows of normalized count data</span>
<span class="va">RIF_input</span> <span class="op">&lt;-</span> <span class="va">Clean_Dat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">Target</span>, <span class="va">TFs</span><span class="op">)</span>, <span class="op">]</span>

<span class="co"># Performing RIF analysis</span>
<span class="va">RIF_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RIF.html">RIF</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">RIF_input</span>,
               nta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">Target</span><span class="op">)</span>,
               ntf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">TFs</span><span class="op">)</span>,
               nSamples1 <span class="op">=</span> <span class="fl">4</span>,
               nSamples2 <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>

<span class="co"># Printing first 10 rows</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">RIF_out</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>bootstrap_options <span class="op">=</span> <span class="st">"striped"</span>, full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<thead><tr>
<th style="text-align:left;">
TF
</th>
<th style="text-align:right;">
avgexpr
</th>
<th style="text-align:right;">
RIF1
</th>
<th style="text-align:right;">
RIF2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000008441
</td>
<td style="text-align:right;">
10.380590
</td>
<td style="text-align:right;">
-0.1386859
</td>
<td style="text-align:right;">
-0.6770894
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000102804
</td>
<td style="text-align:right;">
9.264843
</td>
<td style="text-align:right;">
0.0147001
</td>
<td style="text-align:right;">
0.1789910
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000103888
</td>
<td style="text-align:right;">
13.591598
</td>
<td style="text-align:right;">
0.0985383
</td>
<td style="text-align:right;">
-0.3361795
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000103994
</td>
<td style="text-align:right;">
10.385093
</td>
<td style="text-align:right;">
-1.2748364
</td>
<td style="text-align:right;">
0.8113721
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000112837
</td>
<td style="text-align:right;">
8.143177
</td>
<td style="text-align:right;">
-0.2246323
</td>
<td style="text-align:right;">
-1.7903370
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116016
</td>
<td style="text-align:right;">
11.376028
</td>
<td style="text-align:right;">
-1.2315044
</td>
<td style="text-align:right;">
0.0216295
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116132
</td>
<td style="text-align:right;">
9.388008
</td>
<td style="text-align:right;">
1.4038890
</td>
<td style="text-align:right;">
0.8823288
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000118689
</td>
<td style="text-align:right;">
9.189257
</td>
<td style="text-align:right;">
0.9971123
</td>
<td style="text-align:right;">
0.3981473
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000123562
</td>
<td style="text-align:right;">
9.207009
</td>
<td style="text-align:right;">
0.3381318
</td>
<td style="text-align:right;">
0.4609342
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000157514
</td>
<td style="text-align:right;">
7.215664
</td>
<td style="text-align:right;">
-1.5628941
</td>
<td style="text-align:right;">
-0.3290858
</td>
</tr>
</tbody>
</table>
</div>
<div id="whole-analysis-of-regulatory-impact-factors-rif-and-partial-correlation-and-information-theory-analysis-pcit" class="section level2">
<h2 class="hasAnchor">
<a href="#whole-analysis-of-regulatory-impact-factors-rif-and-partial-correlation-and-information-theory-analysis-pcit" class="anchor"></a>Whole analysis of Regulatory Impact Factors (RIF) and Partial Correlation and Information Theory analysis (PCIT)</h2>
<p>Finally, it is possible to run the entire analysis all at once. The output will be a CeTF object with all results generated between the different steps. To access the CeTF object is recommended to use the accessors from <strong>CeTF</strong> class:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">CeTF</span><span class="op">)</span>

<span class="co"># Loading airway data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span>

<span class="co"># Creating a variable with annotation data</span>
<span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Creating a variable with count data</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>

<span class="co"># Sorting count data samples by conditions (untrt and trt)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span>

<span class="co"># Loading the Transcript Factors (TFs) character</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"TFs"</span><span class="op">)</span>

<span class="co"># Performing the complete analysis</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runAnalysis.html">runAnalysis</a></span><span class="op">(</span>mat <span class="op">=</span> <span class="va">counts</span>, 
                   conditions<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"untrt"</span>, <span class="st">"trt"</span><span class="op">)</span>,
                   lfc <span class="op">=</span> <span class="fl">5</span>,
                   padj <span class="op">=</span> <span class="fl">0.05</span>,
                   TFs <span class="op">=</span> <span class="va">TFs</span>,
                   nSamples1 <span class="op">=</span> <span class="fl">4</span>,
                   nSamples2<span class="op">=</span> <span class="fl">4</span>,
                   tolType <span class="op">=</span> <span class="st">"mean"</span>,
                   diffMethod <span class="op">=</span> <span class="st">"Reverter"</span>, 
                   data.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></code></pre></div>
</div>
<div id="getting-some-graphical-outputs" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-some-graphical-outputs" class="anchor"></a>Getting some graphical outputs</h2>
<p>Based on CeTF class object resulted from <strong>runAnalysis</strong> function is possible to get a plot for Differentially Expressed (DE) genes that shows the relationship between log(baseMean) and Difference of Expression or log2FoldChange, enabling the visualization of the distribution of DE genes and TF in both conditions. These two first plot provides a initial graphical visualization of results:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Using the runAnalysis output (CeTF class object)</span>
<span class="fu"><a href="../reference/SmearPlot.html">SmearPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>, 
          diffMethod <span class="op">=</span> <span class="st">'Reverter'</span>, 
          lfc <span class="op">=</span> <span class="fl">1.5</span>, 
          conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'untrt'</span>, <span class="st">'trt'</span><span class="op">)</span>, 
          type <span class="op">=</span> <span class="st">"DE"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/SmearPlotDE-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Then is possible to generate the same previous plot but now for a single TF. So, if you have a specific TF that you want to visualize, this is the recommended plot:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Using the runAnalysis output (CeTF class object)</span>
<span class="fu"><a href="../reference/SmearPlot.html">SmearPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
          diffMethod <span class="op">=</span> <span class="st">'Reverter'</span>,
          lfc <span class="op">=</span> <span class="fl">1.5</span>,
          conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'untrt'</span>, <span class="st">'trt'</span><span class="op">)</span>,
          TF <span class="op">=</span> <span class="st">'ENSG00000185917'</span>,
          label <span class="op">=</span> <span class="cn">TRUE</span>, 
          type <span class="op">=</span> <span class="st">"TF"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="CeTF_files/figure-html/SmearPlotTF-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The next output is a network for both conditions resulted from :</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Using the runAnalysis output (CeTF class object)</span>
<span class="fu"><a href="../reference/netConditionsPlot.html">netConditionsPlot</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## "none")` instead.

## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## "none")` instead.</code></pre>
<p><img src="CeTF_files/figure-html/netConditionsPlot-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Then we will associate the genes in both conditions with Gene Ontology (GO) and other databases. <em>getGroupGO</em> function will be performed where a set of genes are counted within all possible ontologies, without statistical power. This function is capable of perform groupGO for any organism as long as it’s annotation package exists (i.e. org.Hs.eg.db, org.Bt.eg.db, org.Rn.eg.db, etc). In this example we will perform only for first condition:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading Homo sapiens annotation package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>

<span class="co"># Accessing the network for condition 1</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene1"</span><span class="op">]</span><span class="op">)</span>, 
                  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene2"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Performing getGroupGO analysis</span>
<span class="va">cond1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGroupGO.html">getGroupGO</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">genes</span>, 
                    ont <span class="op">=</span> <span class="st">"BP"</span>, 
                    keyType <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
                    annoPkg <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, 
                    level <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p>Alternatively, it is possible to perform the enrichment analysis with a statistical power. In this case, there are many databases options to perform this analysis (i.e. GO, KEGG, REACTOME, etc). The <em>getEnrich</em> function will perform the enrichment analysis and returns which pathways are enriched in a given set of genes. This analysis requires a reference list of genes with all genes that will be considered to enrich. In this package there is a function, <em>refGenes</em> that has these references genes for ENSEMBL and SYMBOL nomenclatures for 5 organisms: Human (<strong>Homo sapiens</strong>), Mouse (<strong>Mus musculus</strong>), Zebrafish (<strong>Danio rerio</strong>), Cow (<strong>Bos taurus</strong>) and Rat (<strong>Rattus norvegicus</strong>). If the user wants to use a different reference set, simply inputs a character vector with the genes.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Accessing the network for condition 1</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene1"</span><span class="op">]</span><span class="op">)</span>, 
                  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene2"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Performing getEnrich analysis</span>
<span class="va">enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getEnrich.html">getEnrich</a></span><span class="op">(</span>organism<span class="op">=</span><span class="st">'hsapiens'</span>, database<span class="op">=</span><span class="st">'geneontology_Biological_Process'</span>, 
                   genes<span class="op">=</span><span class="va">genes</span>, GeneType<span class="op">=</span><span class="st">'ensembl_gene_id'</span>, 
                   refGene<span class="op">=</span><span class="va">refGenes</span><span class="op">$</span><span class="va">Homo_sapiens</span><span class="op">$</span><span class="va">ENSEMBL</span>, 
                   fdrMethod <span class="op">=</span> <span class="st">"BH"</span>, fdrThr <span class="op">=</span> <span class="fl">0.05</span>, minNum <span class="op">=</span> <span class="fl">5</span>, maxNum <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<p>In the next steps we will use the <em>getGroupGO</em> results, but it is worth mentioning that you can use the results of the <em>getEnrich</em> function. To run the following steps with the <em>getEnrich</em> results is recommended to change the <em>lfc</em> parameter in aboce <em>runAnalysis</em> function by 3. That way there will be more pathways to enrich. Remembering that this will require more processing time.</p>
<p>After the association of GOs and genes, we are able to plot the network of previously results. It’s possible to choose which variables (“pathways,” “TFs” or “genes”) we want to group. All these variables are inputted by the user, i.e.  the user can choose the pathways, TFs or genes returned from analysis from the runAnalysis function and CeTF class object, or can also choose by specific pathways, TFs or genes of interest. The table used to plot the networks are stored in a list resulted from netGOTFPlot function. To get access is needed to use the <em>object$tab</em>. The list will be splitted by groupBy argument choice:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="co"># Subsetting only the first 12 Ontologies with more counts</span>
<span class="va">t1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cond1</span><span class="op">$</span><span class="va">results</span>, <span class="fl">12</span><span class="op">)</span>

<span class="co"># Subsetting the network for the conditions to make available only the 12 nodes subsetted</span>
<span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">cond1</span><span class="op">$</span><span class="va">netGO</span>, <span class="va">cond1</span><span class="op">$</span><span class="va">netGO</span><span class="op">$</span><span class="va">gene1</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">t1</span><span class="op">[</span>, <span class="st">"ID"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>

<span class="co"># generating the GO plot grouping by pathways</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
                    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">as.character</span><span class="op">)</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'pathways'</span>,
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="CeTF_files/figure-html/netGOTFPlot1-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pt</span><span class="op">$</span><span class="va">tab</span><span class="op">$</span><span class="va">`GO:0006807`</span><span class="op">)</span></code></pre></div>
<pre><code>##               gene1    pathway           gene2 class
## 1   ENSG00000004799 GO:0006807 ENSG00000114861  gene
## 2   ENSG00000004799 GO:0006807 ENSG00000119138  gene
## 3   ENSG00000004799 GO:0006807 ENSG00000120093  gene
## 4   ENSG00000004799 GO:0006807 ENSG00000184271  gene
## 49  ENSG00000013293 GO:0019222 ENSG00000114861  gene
## 108 ENSG00000030419 GO:0019222 ENSG00000105989    TF</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># generating the GO plot grouping by TFs</span>
<span class="va">TFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"keytfs"</span><span class="op">)</span><span class="op">$</span><span class="va">TF</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'TFs'</span>,
                  TFs <span class="op">=</span> <span class="va">TFs</span>, 
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="CeTF_files/figure-html/netGOTFPlot1-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pt</span><span class="op">$</span><span class="va">tab</span><span class="op">$</span><span class="va">ENSG00000011465</span><span class="op">)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># generating the GO plot grouping by specifics genes</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ENSG00000011465'</span>, <span class="st">'ENSG00000026025'</span>, <span class="st">'ENSG00000075624'</span>, <span class="st">'ENSG00000077942'</span><span class="op">)</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'genes'</span>,
                  genes <span class="op">=</span> <span class="va">genes</span>,
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="CeTF_files/figure-html/netGOTFPlot1-3.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pt</span><span class="op">$</span><span class="va">tab</span><span class="op">$</span><span class="va">ENSG00000011465</span><span class="op">)</span></code></pre></div>
<pre><code>##              gene1           genes gene2 class
## 1  ENSG00000004799 ENSG00000011465  &lt;NA&gt;  gene
## 5  ENSG00000013293 ENSG00000011465  &lt;NA&gt;  gene
## 9  ENSG00000030419 ENSG00000011465  &lt;NA&gt;    TF
## 13 ENSG00000053254 ENSG00000011465  &lt;NA&gt;    TF
## 17 ENSG00000064961 ENSG00000011465  &lt;NA&gt;    TF
## 21 ENSG00000067082 ENSG00000011465  &lt;NA&gt;    TF</code></pre>
<p>Finally, we will merge the results from the network of condition 1 (Gene-Gene and TF-Gene interaction) with groupGO results (GO-Gene or GO-TF interaction). The final results has the objective to integrate all these data in order to results in a complete view of the data:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
                   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">as.character</span><span class="op">)</span><span class="op">)</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  keyTFs <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"keytfs"</span><span class="op">)</span>, 
                  type <span class="op">=</span> <span class="st">'Integrated'</span><span class="op">)</span>

<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="CeTF_files/figure-html/netGOTFPlot2-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pt</span><span class="op">$</span><span class="va">tab</span><span class="op">)</span></code></pre></div>
<pre><code>##              gene1           gene2
## 22 ENSG00000004799 ENSG00000113761
## 24 ENSG00000004799 ENSG00000114861
## 26 ENSG00000004799 ENSG00000119138
## 27 ENSG00000004799 ENSG00000120093
## 47 ENSG00000004799 ENSG00000143127
## 77 ENSG00000004799 ENSG00000184271</code></pre>
</div>
<div id="using-accessors-to-access-results" class="section level2">
<h2 class="hasAnchor">
<a href="#using-accessors-to-access-results" class="anchor"></a>Using accessors to access results</h2>
<p>Is possible to use the accessors from CeTF class object to access the outputs generated by the <em>runAnalysis</em> function. These outputs can be used as input in <strong>Cytoscape</strong> <span class="citation">(Shannon et al. 2003)</span>. The accessors are:</p>
<ul>
<li>
<strong>getData</strong>: returns the raw, tpm and normalized data;</li>
<li>
<strong>getDE</strong>: returns the DE genes and TFs;</li>
<li>
<strong>InputData</strong> returns the input matrices used to perform RIF and PCIT in <em>runAnalysis</em> function;</li>
<li>
<strong>OutputData</strong> returns the output matrices and lists that output from RIF and PCIT in <em>runAnalysis</em> function;</li>
<li>
<strong>NetworkData</strong> returns the network of interactions between gene and key TFs for both conditions, the keyTFs and the annotation for each gene and TF.</li>
</ul>
<p>All accessors can be further explored by looking in more detail at the documentation.</p>
</div>
</div>
<div id="additional-features" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-features" class="anchor"></a>Additional Features</h1>
<p>This package has some additional features to plot the results. Let’s use the gene set from <em>airway</em> data previously used. The features are:</p>
<div id="network-diffusion-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#network-diffusion-analysis" class="anchor"></a>Network diffusion analysis</h2>
<p>Network propagation is an important and widely used algorithm in systems biology, with applications in protein function prediction, disease gene prioritization, and patient stratification. Propagation provides a robust estimate of network distance between sets of nodes. Network propagation uses the network of interactions to find new genes that are most relevant to a set of well understood genes. To perform this analysis is necessary to install the latest Cytoscape software version and know the path that will be installed the software. After running the diffusion analysis is possible to perform the enrichment for the subnetwork resulted. Finally, ih this vignette we’ll not perform the diffusion analysis because this requires Cytoscape to be installed.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffusion.html">diffusion</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
                    cond <span class="op">=</span> <span class="st">"network1"</span>,
                    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENSG00000185591"</span>, <span class="st">"ENSG00000179094"</span><span class="op">)</span>,
                    cyPath <span class="op">=</span> <span class="st">"C:/Program Files/Cytoscape_v3.7.2/Cytoscape.exe"</span>,
                    name <span class="op">=</span> <span class="st">"top_diffusion"</span>, 
                    label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
</div>
<div id="circos-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#circos-plot" class="anchor"></a>Circos plot</h2>
<p>This plot makes it possible to visualize the targets of specific TFs or genes. The main idea is to identify which chromosomes are linked to the targets of a given gene or TF to infer whether there are cis (same chromosome) or trans (different chromosomes) links between them. The black links are between different chromosomes while the red links are between the same chromosome.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading the CeTF object class demo</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CeTFdemo"</span><span class="op">)</span>

<span class="fu"><a href="../reference/CircosTargets.html">CircosTargets</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">CeTFdemo</span>, 
              file <span class="op">=</span> <span class="st">"/path/to/gtf/file/specie.gtf"</span>, 
              nomenclature <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
              selection <span class="op">=</span> <span class="st">"ENSG00000185591"</span>, 
              cond <span class="op">=</span> <span class="st">"condition1"</span><span class="op">)</span></code></pre></div>
<center>
<img src="fig2.png" style="width:50.0%">
</center>
</div>
<div id="rif-relationships-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#rif-relationships-plots" class="anchor"></a>RIF relationships plots</h2>
<p>To visualize the relationship between RIF results (RIF1 and RIF2) is possible to generate the following plot:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RIFPlot.html">RIFPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
        color  <span class="op">=</span> <span class="st">"darkblue"</span>,
        type   <span class="op">=</span> <span class="st">"RIF"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/RIFPlot1-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Then to get the relationship between RIF1/RIF2 and differential expressed genes:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RIFPlot.html">RIFPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
        color  <span class="op">=</span> <span class="st">"darkblue"</span>,
        type   <span class="op">=</span> <span class="st">"DE"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/RIFPlot2-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="enrichment-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#enrichment-plots" class="anchor"></a>Enrichment plots</h2>
<p>We provide some option to plot the results from <strong>getEnrich</strong> function. First of all we need to perform the enrichment analysis for a set of genes:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Selecting the genes related to the network for condition 1</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene1"</span><span class="op">]</span><span class="op">)</span>, 
                  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene2"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Performing enrichment analysis</span>
<span class="va">cond1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getEnrich.html">getEnrich</a></span><span class="op">(</span>organism<span class="op">=</span><span class="st">'hsapiens'</span>, database<span class="op">=</span><span class="st">'geneontology_Biological_Process'</span>, 
                   genes<span class="op">=</span><span class="va">genes</span>, GeneType<span class="op">=</span><span class="st">'ensembl_gene_id'</span>,
                   refGene<span class="op">=</span><span class="va">refGenes</span><span class="op">$</span><span class="va">Homo_sapiens</span><span class="op">$</span><span class="va">ENSEMBL</span>,
                   fdrMethod <span class="op">=</span> <span class="st">"BH"</span>, fdrThr <span class="op">=</span> <span class="fl">0.05</span>, minNum <span class="op">=</span> <span class="fl">5</span>, maxNum <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<p>After performing the enrichment analysis we have 4 options for viewing this data:</p>
<div id="heatmap-like-functional-classification" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap-like-functional-classification" class="anchor"></a>Heatmap-like functional classification</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/heatPlot.html">heatPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">cond1</span><span class="op">$</span><span class="va">results</span>,
         diff <span class="op">=</span> <span class="fu"><a href="../reference/getDE.html">getDE</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"all"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/heatplot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="circle-barplot" class="section level3">
<h3 class="hasAnchor">
<a href="#circle-barplot" class="anchor"></a>Circle Barplot</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">cond1</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"circle"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/circle-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="barplot" class="section level3">
<h3 class="hasAnchor">
<a href="#barplot" class="anchor"></a>Barplot</h3>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">cond1</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/barplot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="dotplot" class="section level3">
<h3 class="hasAnchor">
<a href="#dotplot" class="anchor"></a>Dotplot</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">cond1</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="CeTF_files/figure-html/dotplot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] dplyr_1.0.7                 org.Hs.eg.db_3.13.0        
##  [3] AnnotationDbi_1.55.1        knitr_1.33                 
##  [5] kableExtra_1.3.4            airway_1.13.0              
##  [7] SummarizedExperiment_1.23.1 Biobase_2.53.0             
##  [9] GenomicRanges_1.45.0        GenomeInfoDb_1.29.3        
## [11] IRanges_2.27.0              S4Vectors_0.31.0           
## [13] BiocGenerics_0.39.1         MatrixGenerics_1.5.1       
## [15] matrixStats_0.60.0          CeTF_1.4.4                 
## 
## loaded via a namespace (and not attached):
##   [1] pbdZMQ_0.3-5                     GGally_2.1.2                    
##   [3] R.methodsS3_1.8.1                clinfun_1.0.15                  
##   [5] coda_0.19-4                      ragg_1.1.3                      
##   [7] tidyr_1.1.3                      ggplot2_3.3.5                   
##   [9] bit64_4.0.5                      DelayedArray_0.19.1             
##  [11] R.utils_2.10.1                   data.table_1.14.0               
##  [13] KEGGREST_1.33.0                  RCurl_1.98-1.3                  
##  [15] doParallel_1.0.16                generics_0.1.0                  
##  [17] cowplot_1.1.1                    RSQLite_2.2.7                   
##  [19] shadowtext_0.0.8                 bit_4.0.4                       
##  [21] tzdb_0.1.2                       enrichplot_1.13.1               
##  [23] base64url_1.4                    webshot_0.5.2                   
##  [25] xml2_1.3.2                       assertthat_0.2.1                
##  [27] viridis_0.6.1                    xfun_0.24                       
##  [29] hms_1.1.0                        jquerylib_0.1.4                 
##  [31] evaluate_0.14                    fansi_0.5.0                     
##  [33] readxl_1.3.1                     igraph_1.2.6                    
##  [35] DBI_1.1.1                        geneplotter_1.71.0              
##  [37] reshape_0.8.8                    GenomicTools.fileHandler_0.1.5.9
##  [39] apcluster_1.4.8                  purrr_0.3.4                     
##  [41] ellipsis_0.3.2                   ggpubr_0.4.0                    
##  [43] backports_1.2.1                  signal_0.7-7                    
##  [45] annotate_1.71.0                  vctrs_0.3.8                     
##  [47] Cairo_1.5-12.2                   abind_1.4-5                     
##  [49] withr_2.4.2                      cachem_1.0.5                    
##  [51] ggforce_0.3.3                    vroom_1.5.3                     
##  [53] sna_2.6                          treeio_1.17.2                   
##  [55] svglite_2.0.0                    cluster_2.1.2                   
##  [57] DOSE_3.19.2                      ape_5.5                         
##  [59] IRdisplay_1.0                    lazyeval_0.2.2                  
##  [61] crayon_1.4.1                     uchardet_1.1.0                  
##  [63] genefilter_1.75.0                labeling_0.4.2                  
##  [65] pkgconfig_2.0.3                  tweenr_1.0.2                    
##  [67] nlme_3.1-152                     rlang_0.4.11                    
##  [69] RJSONIO_1.3-1.4                  lifecycle_1.0.0                 
##  [71] downloader_0.4                   cellranger_1.1.0                
##  [73] rprojroot_2.0.2                  polyclip_1.10-0                 
##  [75] graph_1.71.2                     rngtools_1.5                    
##  [77] Matrix_1.3-4                     aplot_0.0.6                     
##  [79] IRkernel_1.2                     carData_3.0-4                   
##  [81] base64enc_0.1-3                  whisker_0.4                     
##  [83] GlobalOptions_0.1.2              WebGestaltR_0.4.4               
##  [85] png_0.1-7                        viridisLite_0.4.0               
##  [87] rjson_0.2.20                     bitops_1.0-7                    
##  [89] R.oo_1.24.0                      ggnetwork_0.5.10                
##  [91] dplR_1.7.2                       Biostrings_2.61.1               
##  [93] blob_1.2.2                       doRNG_1.8.2                     
##  [95] shape_1.4.6                      stringr_1.4.0                   
##  [97] qvalue_2.25.0                    readr_2.0.0                     
##  [99] rstatix_0.7.0                    gMWT_1.1.1                      
## [101] ggsignif_0.6.2                   scales_1.1.1                    
## [103] memoise_2.0.0                    magrittr_2.0.1                  
## [105] plyr_1.8.6                       zlibbioc_1.39.0                 
## [107] compiler_4.1.0                   scatterpie_0.1.6                
## [109] RColorBrewer_1.1-2               clue_0.3-59                     
## [111] DESeq2_1.33.1                    XVector_0.33.0                  
## [113] patchwork_1.1.1                  MASS_7.3-54                     
## [115] tidyselect_1.1.1                 stringi_1.7.3                   
## [117] forcats_0.5.1                    textshaping_0.3.5               
## [119] highr_0.9                        yaml_2.2.1                      
## [121] GOSemSim_2.19.0                  locfit_1.5-9.4                  
## [123] ggrepel_0.9.1                    grid_4.1.0                      
## [125] sass_0.4.0                       fastmatch_1.1-3                 
## [127] tools_4.1.0                      parallel_4.1.0                  
## [129] rio_0.5.27                       circlize_0.4.13                 
## [131] rstudioapi_0.13                  uuid_0.1-4                      
## [133] foreach_1.5.1                    foreign_0.8-81                  
## [135] gridExtra_2.3                    farver_2.1.0                    
## [137] ggraph_2.0.5                     digest_0.6.27                   
## [139] rvcheck_0.1.8                    snpStats_1.43.0                 
## [141] BiocManager_1.30.16              Rcpp_1.0.7                      
## [143] car_3.0-11                       broom_0.7.9                     
## [145] httr_1.4.2                       ComplexHeatmap_2.9.0            
## [147] colorspace_2.0-2                 rvest_1.0.1                     
## [149] XML_3.99-0.6                     fs_1.5.0                        
## [151] splines_4.1.0                    tidytree_0.3.4                  
## [153] pkgdown_1.6.1                    graphlayouts_0.7.1              
## [155] systemfonts_1.0.2                xtable_1.8-4                    
## [157] jsonlite_1.7.2                   ggtree_3.1.2                    
## [159] tidygraph_1.2.0                  R6_2.5.0                        
## [161] pillar_1.6.2                     htmltools_0.5.1.1               
## [163] glue_1.4.2                       fastmap_1.1.0                   
## [165] clusterProfiler_4.1.1            BiocParallel_1.27.2             
## [167] codetools_0.2-18                 fgsea_1.19.2                    
## [169] mvtnorm_1.1-2                    utf8_1.2.2                      
## [171] lattice_0.20-44                  bslib_0.2.5.1                   
## [173] tibble_3.1.3                     network_1.17.1                  
## [175] curl_4.3.2                       zip_2.2.0                       
## [177] GO.db_3.13.0                     openxlsx_4.2.4                  
## [179] survival_3.2-11                  rmarkdown_2.9                   
## [181] statnet.common_4.5.0             GenomicTools_0.2.9.7            
## [183] repr_1.1.3                       desc_1.3.0                      
## [185] munsell_0.5.0                    DO.db_2.9                       
## [187] GetoptLong_1.0.5                 GenomeInfoDbData_1.2.6          
## [189] iterators_1.0.13                 RCy3_2.13.3                     
## [191] haven_2.4.1                      reshape2_1.4.4                  
## [193] gtable_0.3.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-cover2012elements" class="csl-entry">
Cover, Thomas M, and Joy A Thomas. 2012. <em>Elements of Information Theory</em>. John Wiley &amp; Sons.
</div>
<div id="ref-himes2014rna" class="csl-entry">
Himes, Blanca E, Xiaofeng Jiang, Peter Wagner, Ruoxi Hu, Qiyu Wang, Barbara Klanderman, Reid M Whitaker, et al. 2014. <span>“RNA-Seq Transcriptome Profiling Identifies Crispld2 as a Glucocorticoid Responsive Gene That Modulates Cytokine Function in Airway Smooth Muscle Cells.”</span> <em>PloS One</em> 9 (6): e99625.
</div>
<div id="ref-reverter2008combining" class="csl-entry">
Reverter, Antonio, and Eva KF Chan. 2008. <span>“Combining Partial Correlation and an Information Theory Approach to the Reversed Engineering of Gene Co-Expression Networks.”</span> <em>Bioinformatics</em> 24 (21): 2491–97.
</div>
<div id="ref-reverter2010regulatory" class="csl-entry">
Reverter, Antonio, Nicholas J Hudson, Shivashankar H Nagaraj, Miguel Pérez-Enciso, and Brian P Dalrymple. 2010. <span>“Regulatory Impact Factors: Unraveling the Transcriptional Regulation of Complex Traits from Expression Data.”</span> <em>Bioinformatics</em> 26 (7): 896–904.
</div>
<div id="ref-shannon2003cytoscape" class="csl-entry">
Shannon, Paul, Andrew Markiel, Owen Ozier, Nitin S Baliga, Jonathan T Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker. 2003. <span>“Cytoscape: A Software Environment for Integrated Models of Biomolecular Interaction Networks.”</span> <em>Genome Research</em> 13 (11): 2498–2504.
</div>
<div id="ref-vaquerizas2009census" class="csl-entry">
Vaquerizas, Juan M, Sarah K Kummerfeld, Sarah A Teichmann, and Nicholas M Luscombe. 2009. <span>“A Census of Human Transcription Factors: Function, Expression and Evolution.”</span> <em>Nature Reviews Genetics</em> 10 (4): 252.
</div>
<div id="ref-wang2015regulator" class="csl-entry">
Wang, Kai, and Hiroki Nishida. 2015. <span>“REGULATOR: A Database of Metazoan Transcription Factors and Maternal Factors for Developmental Studies.”</span> <em>BMC Bioinformatics</em> 16 (1): 114.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Carlos Alberto Oliveira de Biagi Junior, Ricardo Perecin 
  Nociti, Breno Osvaldo Funicheli, Wilson Araújo da Silva Junior.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
