<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CeTF application for TCGA-STAD RNA-seq data • CeTF</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CeTF application for TCGA-STAD RNA-seq data">
<meta property="og:description" content="CeTF">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CeTF</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.4.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CeTF.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/TCGA-STAD.html">CeTF application for TCGA-STAD RNA-seq data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TCGA-STAD_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CeTF application for TCGA-STAD RNA-seq data</h1>
                        <h4 class="author">Carlos Biagi Jr</h4>
            
            <h4 class="date">24/01/2020</h4>
      
      
      <div class="hidden name"><code>TCGA-STAD.Rmd</code></div>

    </div>

    
    
<div id="loading-r-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-r-packages" class="anchor"></a>Loading R Packages</h2>
<p>The first step is to load the necessary packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">CeTF</span><span class="op">)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by 'GGally':
##   method from   
##   +.gg   ggplot2</code></pre>
<pre><code>## </code></pre>
<pre><code>## Registered S3 method overwritten by 'ggnetwork':
##   method         from  
##   fortify.igraph ggtree</code></pre>
<pre><code>## ========================================
## CeTF version 1.4.3
## Bioconductor page: http://bioconductor.org/packages/CeTF/
## Github page: https://github.com/cbiagii/CeTF or https://cbiagii.github.io/CeTF/
## Documentation: http://bioconductor.org/packages/CeTF/
## If you use it in published research, please cite:
## 
## Carlos Alberto Oliveira de Biagi Junior, Ricardo Perecin Nociti, Breno Osvaldo 
## Funicheli, Patricia de Cassia Ruy, Joao Paulo Bianchi Ximenez, Wilson A Silva Jr.
## CeTF: an R package to Coexpression for Transcription Factors using Regulatory 
## Impact Factors (RIF) and Partial Correlation and Information (PCIT) analysis.
## bioRxiv. 2020, DOI: 10.1101/2020.03.30.015784
## ========================================</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks">TCGAbiolinks</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span></code></pre></div>
<pre><code>## clusterProfiler v4.0.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/
## 
## If you use clusterProfiler in published research, please cite:
## T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, doi: 10.1016/j.xinn.2021.100141</code></pre>
<pre><code>## 
## Attaching package: 'clusterProfiler'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     filter</code></pre>
</div>
<div id="loading-count-data" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-count-data" class="anchor"></a>Loading count data</h2>
<p>Then we will load the count table for the Stomach adenocarcinoma (TCGA-STAD) data. This data has 35 normal tissue samples (NT) and 415 primary tumor samples (TP).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"TCGA-STAD_Counts_hg19.rda"</span><span class="op">)</span>

<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">exp.matrix</span><span class="op">)</span>

<span class="va">samplesNT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/TCGAquery_SampleTypes.html">TCGAquery_SampleTypes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, <span class="st">"NT"</span><span class="op">)</span>
<span class="va">samplesTP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/TCGAquery_SampleTypes.html">TCGAquery_SampleTypes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, <span class="st">"TP"</span><span class="op">)</span>

<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">samplesNT</span>, <span class="va">samplesTP</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="loading-tfs" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-tfs" class="anchor"></a>Loading TFs</h2>
<p>The list of TFs for <em>Homo sapies</em> is provided in CeTF package. Here, we will convert from ENSEMBL to SYMBOL nomenclature.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">TFs</span><span class="op">)</span>
<span class="va">convert</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html">bitr</a></span><span class="op">(</span><span class="va">TFs</span>, fromType <span class="op">=</span> <span class="st">"ENSEMBL"</span>,
                toType <span class="op">=</span> <span class="st">"SYMBOL"</span>,
                OrgDb <span class="op">=</span> <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html">org.Hs.eg.db</a></span><span class="op">)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(TFs, fromType = "ENSEMBL", toType = "SYMBOL", OrgDb =
## org.Hs.eg.db::org.Hs.eg.db): 0.17% of input gene IDs are fail to map...</code></pre>
</div>
<div id="performing-runanalysis-function" class="section level2">
<h2 class="hasAnchor">
<a href="#performing-runanalysis-function" class="anchor"></a>Performing runAnalysis function</h2>
<p>Finally, will be performed the complete analysis for this data</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runAnalysis.html">runAnalysis</a></span><span class="op">(</span>mat <span class="op">=</span> <span class="va">counts</span>, 
                   conditions<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NT"</span>, <span class="st">"TP"</span><span class="op">)</span>,
                   lfc <span class="op">=</span> <span class="fl">2.57</span>,
                   padj <span class="op">=</span> <span class="fl">0.05</span>,
                   TFs <span class="op">=</span> <span class="va">convert</span><span class="op">$</span><span class="va">SYMBOL</span>,
                   nSamples1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">samplesNT</span><span class="op">)</span>,
                   nSamples2<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">samplesTP</span><span class="op">)</span>,
                   tolType <span class="op">=</span> <span class="st">"mean"</span>,
                   diffMethod <span class="op">=</span> <span class="st">"Reverter"</span>, 
                   data.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></code></pre></div>
</div>
<div id="checking-the-data-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-the-data-distribution" class="anchor"></a>Checking the data distribution</h2>
<p>Before proceeding to view the results, we will check the data distribution:</p>
<div id="histogram-of-connectivity-distribution" class="section level4">
<h4 class="hasAnchor">
<a href="#histogram-of-connectivity-distribution" class="anchor"></a>Histogram of connectivity distribution</h4>
<p>It will be generated a histogram for adjacency matrix to show the clustering coefficient distribution for both conditions (NT and TP).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For NT condition</span>
<span class="fu"><a href="../reference/histPlot.html">histPlot</a></span><span class="op">(</span><span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit1"</span>, <span class="st">"adj_sig"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution1-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For TP condition</span>
<span class="fu"><a href="../reference/histPlot.html">histPlot</a></span><span class="op">(</span><span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit2"</span>, <span class="st">"adj_sig"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution2-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="density-distribution-of-correlation-coefficients-and-significant-pcit-values" class="section level4">
<h4 class="hasAnchor">
<a href="#density-distribution-of-correlation-coefficients-and-significant-pcit-values" class="anchor"></a>Density distribution of correlation coefficients and significant PCIT values</h4>
<p>It will be generated a density plot for adjacency matrices using the raw adjacency matrix and significant adjacency matrix resulted from PCIT function for both conditions (NT and TP).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For NT condition</span>
<span class="fu"><a href="../reference/densityPlot.html">densityPlot</a></span><span class="op">(</span>mat1 <span class="op">=</span> <span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit1"</span>, <span class="st">"adj_raw"</span><span class="op">)</span>, 
            mat2 <span class="op">=</span> <span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit1"</span>, <span class="st">"adj_sig"</span><span class="op">)</span>, 
            threshold <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution3-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For TP condition</span>
<span class="fu"><a href="../reference/densityPlot.html">densityPlot</a></span><span class="op">(</span>mat1 <span class="op">=</span> <span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit2"</span>, <span class="st">"adj_raw"</span><span class="op">)</span>, 
            mat2 <span class="op">=</span> <span class="fu"><a href="../reference/OutputData.html">OutputData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"pcit2"</span>, <span class="st">"adj_sig"</span><span class="op">)</span>, 
            threshold <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="relationship-plots-between-rif1-rif2-and-de-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#relationship-plots-between-rif1-rif2-and-de-genes" class="anchor"></a>Relationship plots between RIF1, RIF2 and DE genes</h4>
<p>It will be generated plots for the relationship between the RIF output analysis (RIF1 and RIF2) and for differentially expressed genes (DE).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RIFPlot.html">RIFPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
        type   <span class="op">=</span> <span class="st">"RIF"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution5-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RIFPlot.html">RIFPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
        type   <span class="op">=</span> <span class="st">"DE"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/distribution6-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="smear-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#smear-plot" class="anchor"></a>Smear Plot</h2>
<p>After checking data distribution, is possible to visualize specific results. The first one is the smear plot, that corresponds to a plot for Differentially Expressed (DE) genes and for specific TF that shows the relationship between log(baseMean) and Difference of Expression or log2FoldChange.</p>
<div id="differentially-expressed-genes-degs" class="section level4">
<h4 class="hasAnchor">
<a href="#differentially-expressed-genes-degs" class="anchor"></a>Differentially Expressed Genes (DEGs)</h4>
<p>The first option allows the visualization of all DEGs in both conditions.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SmearPlot.html">SmearPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>, 
          diffMethod <span class="op">=</span> <span class="st">'Reverter'</span>, 
          lfc <span class="op">=</span> <span class="fl">2</span>, 
          conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'NT'</span>, <span class="st">'TP'</span><span class="op">)</span>, 
          type <span class="op">=</span> <span class="st">"DE"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/smear1-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="specific-tfs" class="section level4">
<h4 class="hasAnchor">
<a href="#specific-tfs" class="anchor"></a>Specific TFs</h4>
<p>The second option allows the visualization of a specific TF of interest showing how its targets are distributed. The table with keyTFs can be accessed using <strong>NetworkData(out, “keytfs”)</strong> code. In this example we will choose the SMAD2 gene.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/SmearPlot.html">SmearPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
          diffMethod <span class="op">=</span> <span class="st">'Reverter'</span>,
          lfc <span class="op">=</span> <span class="fl">2</span>,
          conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'NT'</span>, <span class="st">'TP'</span><span class="op">)</span>,
          TF <span class="op">=</span> <span class="st">'SETD3'</span>,
          label <span class="op">=</span> <span class="cn">FALSE</span>, 
          type <span class="op">=</span> <span class="st">"TF"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/smear2-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="network-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#network-visualization" class="anchor"></a>Network visualization</h2>
<p>The package also allows visualization of the interaction networks for both conditions. Remembering that there is the possibility to access the tables of both networks using <strong>NetworkData(out, “network1”) or NetworkData(out, “network2”)</strong> and use Cytoscape to personalize the view of the networks.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/netConditionsPlot.html">netConditionsPlot</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## "none")` instead.

## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =
## "none")` instead.</code></pre>
<p><img src="TCGA-STAD_files/figure-html/netvis-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="network-genes-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#network-genes-enrichment" class="anchor"></a>Network genes enrichment</h2>
<p>After obtaining the interaction networks, it is possible to perform the functional enrichment of these genes for both conditions. In this example, we will perform enrichment only for the NT condition genes. Remembering that there is the possibility of performing a grouping of genes, better known as group GO, where only the genes that form part of the pathways will be counted, with no statistical inference. The <strong>getGroupGO</strong> function is used for this:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Loading Homo sapiens annotation package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:clusterProfiler':
## 
##     rename</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## 
## Attaching package: 'IRanges'</code></pre>
<pre><code>## The following object is masked from 'package:clusterProfiler':
## 
##     slice</code></pre>
<pre><code>## 
## Attaching package: 'AnnotationDbi'</code></pre>
<pre><code>## The following object is masked from 'package:clusterProfiler':
## 
##     select</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Selecting genes only for the NT condition</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene1"</span><span class="op">]</span><span class="op">)</span>, 
                  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span><span class="op">[</span>, <span class="st">"gene2"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getEnrich.html">getEnrich</a></span><span class="op">(</span>organism<span class="op">=</span><span class="st">'hsapiens'</span>, database<span class="op">=</span><span class="st">'geneontology_Biological_Process'</span>, 
                    genes<span class="op">=</span><span class="va">genes</span>, GeneType<span class="op">=</span><span class="st">'genesymbol'</span>, 
                    refGene<span class="op">=</span><span class="va">refGenes</span><span class="op">$</span><span class="va">Homo_sapiens</span><span class="op">$</span><span class="va">SYMBOL</span>, 
                    fdrMethod <span class="op">=</span> <span class="st">'BH'</span>, fdrThr <span class="op">=</span> <span class="fl">0.05</span>, minNum <span class="op">=</span> <span class="fl">5</span>, maxNum <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="visualization-of-functional-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization-of-functional-enrichment" class="anchor"></a>Visualization of functional enrichment</h2>
<p>This package provides four different types of visualization for functional enrichment results:</p>
<div id="heatmap-like-functional-classification" class="section level4">
<h4 class="hasAnchor">
<a href="#heatmap-like-functional-classification" class="anchor"></a>Heatmap-like functional classification</h4>
<p>Enables the visualization of the enrichment analysis results from getEnrich function. The plot contains the heatmap with the associated pathways genes, the significance of the enrichment and a barplot with the enrichment ratio.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/heatPlot.html">heatPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">enrich</span><span class="op">$</span><span class="va">results</span>,
         diff <span class="op">=</span> <span class="fu"><a href="../reference/getDE.html">getDE</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"all"</span><span class="op">)</span>, 
         showCategory <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/enrich2-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="circular-barplot" class="section level4">
<h4 class="hasAnchor">
<a href="#circular-barplot" class="anchor"></a>Circular barplot</h4>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">enrich</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"circle"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/enrich3-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="barplot" class="section level4">
<h4 class="hasAnchor">
<a href="#barplot" class="anchor"></a>Barplot</h4>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">enrich</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/enrich4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="dotplot" class="section level4">
<h4 class="hasAnchor">
<a href="#dotplot" class="anchor"></a>Dotplot</h4>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/enrichPlot.html">enrichPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">enrich</span><span class="op">$</span><span class="va">results</span>,
           type <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/enrich5-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="network-for-ontologies-genes-and-tfs" class="section level2">
<h2 class="hasAnchor">
<a href="#network-for-ontologies-genes-and-tfs" class="anchor"></a>Network for Ontologies, genes and TFs</h2>
<p>Generates the plot of groupGO or getEnrch network result of <strong>getGroupGO</strong> or <strong>getEnrich</strong> function, and the integrated network of genes, GOs and TFs.</p>
<div id="grouping-by-pathways" class="section level4">
<h4 class="hasAnchor">
<a href="#grouping-by-pathways" class="anchor"></a>Grouping by Pathways</h4>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">t1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">enrich</span><span class="op">$</span><span class="va">results</span>, <span class="fl">15</span><span class="op">)</span>
<span class="va">t2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">enrich</span><span class="op">$</span><span class="va">netGO</span>, <span class="va">enrich</span><span class="op">$</span><span class="va">netGO</span><span class="op">$</span><span class="va">gene1</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">t1</span><span class="op">[</span>, <span class="st">"ID"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'pathways'</span>,
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/netGOTFPlot1-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="grouping-by-tfs" class="section level4">
<h4 class="hasAnchor">
<a href="#grouping-by-tfs" class="anchor"></a>Grouping by TFs</h4>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">TFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ECD"</span>, <span class="st">"SOX4"</span>, <span class="st">"TERF1"</span>, <span class="st">"TP53"</span><span class="op">)</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'TFs'</span>,
                  TFs <span class="op">=</span> <span class="va">TFs</span>, 
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/netGOTFPlot2-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="grouping-by-tfs-1" class="section level4">
<h4 class="hasAnchor">
<a href="#grouping-by-tfs-1" class="anchor"></a>Grouping by TFs</h4>
<p>It is possible to choose for specif genes to visualize how them are related with the enriched pathways and TFs.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CDH1'</span>, <span class="st">'TP53'</span><span class="op">)</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  resultsGO <span class="op">=</span> <span class="va">t1</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  anno <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"annotation"</span><span class="op">)</span>,
                  groupBy <span class="op">=</span> <span class="st">'genes'</span>,
                  genes <span class="op">=</span> <span class="va">genes</span>,
                  type <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/netGOTFPlot3-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="integrated" class="section level4">
<h4 class="hasAnchor">
<a href="#integrated" class="anchor"></a>Integrated</h4>
<p>It is also possible to integrate the TFs, genes and Pathways in the same network:</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/netGOTFPlot.html">netGOTFPlot</a></span><span class="op">(</span>netCond <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"network1"</span><span class="op">)</span>,
                  netGO <span class="op">=</span> <span class="va">t2</span>,
                  keyTFs <span class="op">=</span> <span class="fu"><a href="../reference/NetworkData.html">NetworkData</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"keytfs"</span><span class="op">)</span>, 
                  type <span class="op">=</span> <span class="st">'Integrated'</span><span class="op">)</span>
<span class="va">pt</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="TCGA-STAD_files/figure-html/netGOTFPlot4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="circos-targets" class="section level2">
<h2 class="hasAnchor">
<a href="#circos-targets" class="anchor"></a>Circos Targets</h2>
<p>This plot makes it possible to visualize the targets of specific TFs or genes. The main idea is to identify which chromosomes are linked to the targets of a given gene or TF to infer whether there are cis (same chromosome) or trans (different chromosomes) links between them. The black links are between different chromosomes while the red links are between the same chromosome.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CircosTargets.html">CircosTargets</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>, 
              file <span class="op">=</span> <span class="st">"/path/to/gtf/file.gtf"</span>, 
              nomenclature <span class="op">=</span> <span class="st">"SYMBOL"</span>, 
              selection <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SSRP1"</span>, <span class="st">"MAF"</span>, <span class="st">"SETD3"</span>, <span class="st">"HOXB3"</span><span class="op">)</span>, 
              cond <span class="op">=</span> <span class="st">"condition1"</span><span class="op">)</span></code></pre></div>
</div>
<div id="diffusion-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#diffusion-analysis" class="anchor"></a>Diffusion Analysis</h2>
<p>Network propagation is an important and widely used algorithm in systems biology, with applications in protein function prediction, disease gene prioritization, and patient stratification. Propagation provides a robust estimate of network distance between sets of nodes. Network propagation uses the network of interactions to find new genes that are most relevant to a set of well understood genes. To perform this analysis is necessary to install the latest Cytoscape software version and know the path that will be installed the software. After running the diffusion analysis is possible to perform the enrichment for the subnetwork resulted.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffusion.html">diffusion</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">out</span>,
                    cond <span class="op">=</span> <span class="st">"network1"</span>,
                    genes <span class="op">=</span> <span class="st">"HOXB3"</span>,
                    cyPath <span class="op">=</span> <span class="st">"/path/to/cytoscape"</span>,
                    name <span class="op">=</span> <span class="st">"top_diffusion"</span>, 
                    label <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">result</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Carlos Alberto Oliveira de Biagi Junior, Ricardo Perecin 
  Nociti, Breno Osvaldo Funicheli, Wilson Araújo da Silva Junior.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
